docs/USAGE.md

```diff
@@ -85,3 +85,41 @@ java -jar gtfs-validator-SNAPSHOT.jar --export_notice_schema --url https://url/t
 Please note that since downloading will take time, we recommend validating repeatedly on a local file.
+
+## via spring boot app
+
+### Launch the service
+1. Install [Java 8 or higher](https://www.oracle.com/java/technologies/javase-downloads.html).
+1. Download the latest gtfs-validator_springboot JAR file from our [Releases page](https://github.com/MobilityData/gtfs-validator/releases) or snapshot artifact from [GitHub Actions](https://github.com/MobilityData/gtfs-validator/actions?query=branch%3Amaster).
+1. Authenticate to Google Storage 
+Run the following command line: 
+```
+java -jar gtfs-validator-v2.1.0_springboot.jar
+```
+
+After launching the jar of the springboot application, the validator will be available as a service on port 8080. 
+
+**Full list of query parameters available**
+
+| Query parameter             	| required? 	| Description                                                                                      	| Default value        	|
+|-----------------------------	|-----------	|--------------------------------------------------------------------------------------------------	|----------------------	|
+| `output_base`               	| Optional  	| Base directory to store the outputs.                                                             	| "output"             	|
+| `threads`                   	| Optional  	| Number of threads to use.                                                                        	| 8                    	|
+| `country_code`              	| Optional  	| Country code of the feed, e.g., `nl`. It must be a two-letter country code (ISO 3166-1 alpha-2). 	| null                 	|
+| `url`                       	| Required  	| Fully qualified URL to download GTFS archive.                                                    	|                      	|
+| `validation_report_name`    	| Optional  	| Name of the validation report (including `.json` extension).                                     	| "report.json"        	|
+| `system_errors_report_name` 	| Optional  	| Name of the system errors report (including `.json` extension).                                  	| "system_errors.json" 	|
+| `dataset_id`                	| Optional  	| The id of the dataset.                                                                           	| "dataset id value"  	|
+| `commit_sha`                	| Optional  	| The SHA of the commit from Github.                                                               	| "commit sha value"  	|
+
+Sample query:
+
+```
+http://localhost:8080/?url=http://webapps.thebus.org/transitdata/Production/google_transit.zip&dataset_id=id_value&commit_sha=sha_value
+```
+
+which will:
+1. execute the validator on the dataset available at: http://webapps.thebus.org/transitdata/Production/google_transit.zip
+1. store the validation report as report.json in the bucket specified by environment variable `VALIDATION_REPORT_BUCKET` sha_value/id_value.
+
```

Comment:
> The documentation above is great to launch the spring boot app locally. Is the documentation about deploying it on Kubernetes is planned in this PR? I think it would be a great addition to document how
> 
> 1. the app can be deployed on Kubernetes;
> 2. the pipeline from having a list of URLs to storing the reports in Google Cloud Storage, passing by the spring boot app, docker and Kubernetes;
> 3. a quick explanation on why spring boot was necessary.
> 
> I think your [architecture](https://github.com/MobilityData/gtfs-validator/pull/933#issuecomment-902075651) is a really good start for points 2 and 3.

I was not planning on including the documentation to deploy to Kubernetes within this PR but in https://github.com/MobilityData/gtfs-validator/pull/848 - However, given your feedback, I will include documentation that explains why this springboot layer is needed to run the acceptance tests on Kubernetes.

Reply:
> Thanks @lionel-nj for putting [the architecture diagram](https://github.com/MobilityData/gtfs-validator/pull/933#issuecomment-902075651) together! I agree with @maximearmstrong, I'd put it in the documentation in this PR directly so it's included in the project.
> 
> However, I had a hard time deciphering exactly what was happening looking at the current architecture diagram. I think you're mixing architecture with process diagrams here which I found confusing. The GitHub Action shading encompasses the entire data flow, which makes it seem like everything is running on GitHub's infrastructure (it's not, right? The actual validation runs as the Spring Boot application on Google Cloud infrastructure?).

You're right, the Github action is the starting point of the pipeline: but the operations do not run in the Github Actions infrastructure.

> My understanding is that the overall architecture logical data flow looks like the following:
> 
> * GitHub Action (Validator is compiled) -> Kubernetes on Google Cloud (Validators are executed) -> GitHub Action (Output files from executed validators are compared)
> 
> Is that right?

Absolutely!

> It would also be helpful to label the parts of the architecture with what is completed (i.e., in this PR or already exists) and what is not yet implemented).

Will do.